File: matrixplot.m
%% Dados
clear all
clc
close all
load('metaInputs.mat');

%Choosing only 4 parameters of the news (the ones that seem more important)
metaInputs2 = metaInputs([1 2 3 19],:);

metaInputsTrue = metaInputs2(:,[1:3602]);
metaInputsFalse = metaInputs2(:,[3603:end]);

figure (1)
subplot(1,2,1)
plotmatrix(metaInputsTrue') 
title('True news')

hold on
subplot(1,2,2)
plotmatrix(metaInputsFalse')
title('Fake news')

File: matrixplot.py
import numpy as np
import scipy.io
import matplotlib.pyplot as plt

# Load metaInputs from the .mat file
mat_data = scipy.io.loadmat('metaInputs.mat')

# Access the metaInputs data
metaInputs = mat_data['metaInputs']

# Choosing only 4 parameters of the news
metaInputs2 = metaInputs[[0, 1, 2, 18], :]

metaInputsTrue = metaInputs2[:, :3602]
metaInputsFalse = metaInputs2[:, 3602:]

plt.figure(1)

plt.subplot(1, 2, 1)
plt.plot(metaInputsTrue.T)
plt.title('True news')

plt.subplot(1, 2, 2)
plt.plot(metaInputsFalse.T)
plt.title('Fake news')

plt.tight_layout()
plt.show()
